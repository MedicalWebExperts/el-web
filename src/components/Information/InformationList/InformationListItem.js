import React from 'react';
import { string, shape } from 'prop-types';
import css from 'styled-jsx/css';

import theme from '../../../styles';

const propTypes = {
  text: string.isRequired,
  styles: shape({}),
};

const defaultProps = {
  styles: {},
};

const style = css`
  li {
    padding: 0 0 0 20px;
    position: relative;
    color: ${theme.colors.default};
    font-family: ${theme.font.primary};
    font-size: ${theme.font.size}px;
    line-height: 1.6;
  }
  li:before {
    content: '';
    position: absolute;
    left: 0;
    top: 7px;
    right: auto;
    width: 8px;
    height: 8px;
    border-bottom: solid 1px currentColor;
    border-left: solid 1px currentColor;
    transform: rotate(-135deg);
    color: #4a90e2;
  }
`;

const InformationListItem = ({ text, styles }) => (
  <li style={styles}>
    <style jsx>{style}</style>
    {text}
  </li>
);

InformationListItem.propTypes = propTypes;
InformationListItem.defaultProps = defaultProps;
export default InformationListItem;
