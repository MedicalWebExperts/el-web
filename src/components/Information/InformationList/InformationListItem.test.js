import React from 'react';
import renderer from 'react-test-renderer';
import InformationListItem from './InformationListItem';

const defaultProps = {
  text: 'InformationListItem test',
};

const customStyles = {
  styles: { padding: 20 },
};

describe('Component InformationListItem Snapshot', () => {
  it('without arguments, take the default values', () => {
    const tree = renderer.create(<InformationListItem {...defaultProps} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('using custom styles', () => {
    const tree = renderer
      .create(<InformationListItem {...Object.assign({}, defaultProps, customStyles)} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
