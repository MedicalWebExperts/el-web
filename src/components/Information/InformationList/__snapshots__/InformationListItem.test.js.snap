// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component InformationListItem Snapshot using custom styles 1`] = `
<li
  className="jsx-1051720404"
  style={
    Object {
      "padding": 20,
    }
  }
>
  InformationListItem test
</li>
`;

exports[`Component InformationListItem Snapshot without arguments, take the default values 1`] = `
<li
  className="jsx-1051720404"
  style={Object {}}
>
  InformationListItem test
</li>
`;
